adapters/faraday/lib/opentelemetry/adapters/faraday/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Faraday
+      module Middlewares
+        class TracerMiddleware < ::Faraday::Middleware
+          def call(env)
+            tracer.in_span(env.url.to_s,
+                          attributes: { 'component' => 'http',
```

Comment:
We'll want to go through the [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md) with a fine-toothed comb.

Reply:
It looks like for the client, that only `http.url` is required.